authorization/v2/uniformResourceIdentifier_test.go

```diff
@@ -224,3 +224,59 @@ func Test_UnmarshalJSON__should_return_correct_URI(t *testing.T) {
 
 	assert.Equal(t, identifier, expectedURI)
 }
+
+func Test_CompareURI__should_return_true_with_source_in_target_set(t *testing.T) {
```

Comment:
I'd say it'd be better to run these granular tests against `isURIMatch` rather than `CompareURI` because we're testing the functionality of `isURIMatch` not `CompareURI`. Then we could have a very simple test for `CompareURI` which just tests a single source URI with multiple target URIs.